### DISCUSSION TITLE
关于配置的问题

### DISCUSSION THREAD

**Discussion by reinalin** [cgerchenhp/UE_TAPython_Plugin_Release#18] [General]
1. 公司这边工程创建的很频繁, 可能每个制作者每天要创建好几个工程, 每次都要将插件拷贝到工程plugins下显得很不方便
2. 如果放在engine plugins下时,  我想把 "<工程>\TA\TAPython\Config\Config.ini" 中的MenuConfigFilePath, ChameleonDataFolder, PythonContentFolder三个参数换成服务器的路径方便集中更新TA\TAPython下的内容, 但替换服务器路径后并不能生效, 我想这个参数的路径可能是插件初始化时在前面添加了工程的路径(因为现在就是个相对路径)

这是目前碰到的配置的痛点

**Comment by cgerchenhp** [comment]
设置中上述的三个路径是工程中的相对路径，因此会失效。

关于“拷贝插件和已有工具”的问题，也许可以考虑用类似下面的方式简化，并且方便工具能在在多个工程中公用。

### 方法一：
每次创建新的工程后，在新工程的目录中，执行下面cmd命令，或者执行一个bat批处理。用软链接的方式共享

```
mkdir Plugins
mklink /d .\Plugins\TAPython \\168.xx.xx.xx\SomeWhere\TAPythonPlugin
mklink /d .\TA \\168.xx.xx.xx\SomeWhere\TA
```
这样就将服务器中目录中的TAPython插件，和TA目录链接到了本地工程下。


### 方法二：
1. 服务器端放置TAPython插件，并在其Resources/DefaultResource/TA/TAPython中添加一个同步服务器中已有TAPython工具的工具
2. 通过```mklink /d .\Plugins\TAPython \\168.xx.xx.xx\SomeWhere\TAPythonPlugin```将服务器中的目录链接到本地
3. 打开新工程，设置AdditionalPaths后，执行步骤1中的同步工具
这样同步之后的TA目录依然是本地目录，多个工程直接不会冲突

**Reply by VXeffect** [reply]
大佬，我设置了插件，界面显示了，但是提示路径不对，说NeedAddPythonPath，可是我找不到教程里的Python文件夹了呀

**Comment by cgerchenhp** [comment]
需要在Project Settings / Plugins / Python - 中添加路径，设置后需要重启编辑器。这里有个视频，可以按照步骤设置一下：https://youtu.be/ycWDLrc5fDg 。可以先用一个空的工程，在不修改任何配置的情况下试验。

**Comment by cgerchenhp** [comment]
在v1.2.0中，已经支持在"MenuConfigFilePath"和"PythonContentFolder"中使用绝对路径，用于本地的各UE工程共同使用同一个目录中的工具。（"ChameleonDataFolder"实际已几乎被废弃，不用设置）
